go-controller/pkg/ovn/address_set_sync.go

```diff
@@ -0,0 +1,447 @@
+	// stale address sets don't have controller ID
+	p := libovsdbops.GetNoOwnerPredicate[*nbdb.AddressSet]()
+	addrSetList, err := libovsdbops.FindAddressSetsWithPredicate(syncer.nbClient, p)
+	if err != nil {
+		return fmt.Errorf("failed to find stale address sets: %v", err)
+	}
+	updatedCount := 0
+	defer func() {
+		klog.Infof("SyncAddressSets updated %d of %d stale address sets", updatedCount, len(addrSetList))
+	}()
+	lastBatch := len(addrSetList) / syncer.txnBatchSize
+	if len(addrSetList)%syncer.txnBatchSize > 0 {
+		lastBatch += 1
+	}
+	for batchIdx := 0; batchIdx < lastBatch; batchIdx++ {
+		maxIdx := (batchIdx + 1) * syncer.txnBatchSize
+		if len(addrSetList) < maxIdx {
+			maxIdx = len(addrSetList)
+		}
+		for addrSetIdx := batchIdx * syncer.txnBatchSize; addrSetIdx < maxIdx; addrSetIdx++ {
+			if err = syncer.updateOpsForAS(addrSetList[addrSetIdx]); err != nil {
+				return err
+			}
+		}
+		// update refs
+		aclsToUpdate := []*nbdb.ACL{}
+		for _, replaceRef := range syncer.replaceRefACLs {
+			acl := replaceRef.dbobj
+			for oldHash, newHash := range replaceRef.replace {
+				acl.Match = strings.ReplaceAll(acl.Match, "$"+oldHash, "$"+newHash)
+			}
+			aclsToUpdate = append(aclsToUpdate, acl)
+		}
+		ops, err := libovsdbops.CreateOrUpdateACLsOps(syncer.nbClient, nil, aclsToUpdate...)
+		if err != nil {
+			return fmt.Errorf("failed to get update acl ops: %v", err)
+		}
+
+		qosesToUpdate := []*nbdb.QoS{}
+		for _, replaceRef := range syncer.replaceRefQoSes {
+			qos := replaceRef.dbobj
+			for oldHash, newHash := range replaceRef.replace {
+				qos.Match = strings.ReplaceAll(qos.Match, "$"+oldHash, "$"+newHash)
+			}
+			qosesToUpdate = append(qosesToUpdate, qos)
+		}
+		ops, err = libovsdbops.CreateOrUpdateQoSesOps(syncer.nbClient, ops, qosesToUpdate...)
+		if err != nil {
+			return fmt.Errorf("failed to get update qos ops: %v", err)
+		}
+
+		lrpsToUpdate := []*nbdb.LogicalRouterPolicy{}
+		for _, replaceRef := range syncer.replaceRefLRPs {
+			lrp := replaceRef.dbobj
+			for oldHash, newHash := range replaceRef.replace {
+				lrp.Match = strings.ReplaceAll(lrp.Match, "$"+oldHash, "$"+newHash)
+			}
+			lrpsToUpdate = append(lrpsToUpdate, lrp)
+		}
+		ops, err = libovsdbops.CreateOrUpdateLogicalRouterPoliciesOps(syncer.nbClient, ops, ovntypes.OVNClusterRouter, lrpsToUpdate...)
+		if err != nil {
+			return fmt.Errorf("failed to get update LRPs ops: %v", err)
+		}
+		syncer.ops = append(syncer.ops, ops...)
+
+		// delete stale address sets
+		for asName, staleAS := range syncer.staleASOps {
+			if syncer.newASs[asName] {
+				// newer version for the same address set is found, delete old
+				syncer.ops = append(syncer.ops, staleAS.deleteOps...)
+			} else {
+				// no newer version, update stale AS
+				syncer.ops = append(syncer.ops, staleAS.updateOps...)
+			}
+		}
+		_, err = libovsdbops.TransactAndCheck(syncer.nbClient, syncer.ops)
+		if err != nil {
+			return fmt.Errorf("failed to transact address set sync ops: %v", err)
```

Comment:
we could get a giant output here depending on how many ops were executed:
```
	if err != nil {
		return nil, fmt.Errorf("error in transact with ops %+v results %+v and errors %+v: %v", ops, results, opErrors, err)
	}
```

not saying you need to change anything, just pointing it out

Reply:
true, but I guess since this will kill the whole process anyway, it's probably ok?